File: android/kit/src/test/kotlin/space/dodoplanet/gildongmu/kit/TransitAlternativeNameTest.kt
package space.dodoplanet.gildongmu.kit

import kotlinx.serialization.Serializable
import kotlin.test.Test
import kotlin.test.assertEquals
import kotlin.test.assertTrue

/**
 * 대안 경로 이름 조립 규칙(spec 2026-09-24 §4.1). 웹 `transit-alternative-name.test.ts`·Kit
 * `TransitAlternativeNameTests`와 같은 공유 fixture(`transit-alternative-name-cases.json`)를 읽는다.
 */
class TransitAlternativeNameTest {
    @Serializable
    private data class CaseFile(val cases: List<Case>) {
        @Serializable
        data class Case(
            val name: String,
            val highlight: List<String>? = null,
            val displayIndex: Int? = null,
            val parts: List<Part>,
        )

        @Serializable
        data class Part(val key: String, val index: Int? = null)
    }

    @Test fun sharedCases() {
        val cases = Fixtures.sharedJson("transit-alternative-name-cases.json", CaseFile.serializer()).cases
        assertTrue(cases.size >= 10)
        for (c in cases) {
            val got = TransitAlternativeName.parts(c.highlight, c.displayIndex).map { CaseFile.Part(it.key, it.index) }
            assertEquals(c.parts, got, c.name)
        }
    }
}

File: src/lib/__tests__/transit-alternative-name.test.ts
import { describe, expect, it } from "vitest";
import cases from "./fixtures/transit-alternative-name-cases.json";
import { alternativeName, alternativeNameParts } from "../transit-alternative-name";
import type { TransitHighlight } from "../types";

// 공유 fixture — Kit `TransitAlternativeNameTests`·`:kit` `TransitAlternativeNameTest`가 같은 표를 읽는다.
describe("alternativeNameParts (공유 fixture)", () => {
  it.each(cases.cases.map((c) => [c.name, c] as const))("%s", (_, c) => {
    const parts = alternativeNameParts({
      highlight: (c.highlight ?? undefined) as TransitHighlight[] | undefined,
      displayIndex: c.displayIndex ?? undefined,
    });
    expect(parts.map((p) => ({ key: `route.transit.${p.key}`, index: (p.values.index as number | undefined) ?? null }))).toEqual(c.parts);
  });
});

describe("alternativeName", () => {
  it("조각을 쉼표로 이어 한 줄로 만든다(가운뎃점 없음)", () => {
    const name = alternativeName({ highlight: ["busOnly", "fewestTransfers"] }, (key) => `<${key}>`);
    expect(name).toBe("<alternativeFewestTransfers>, <alternativeBusOnly>");
  });
});
